pkgs/tools/X11/caffeine-ng/default.nix

```diff
@@ -1,23 +1,25 @@
 { gdk-pixbuf, glib, gobject-introspection, gtk3, lib, libnotify,
-  pythonPackages, wrapGAppsHook
+  python3Packages, wrapGAppsHook
 }:
 
-pythonPackages.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "caffeine-ng";
   version = "3.4.2";
 
-  src = pythonPackages.fetchPypi{
+  src = python3Packages.fetchPypi{
     inherit pname version;
     sha256="05k8smjlfjcccgmp8qi04l7106k46fs4p8fl5bdqqjwv6pwl7y4w";
   };
 
   nativeBuildInputs = [ wrapGAppsHook glib ];
   buildInputs = [ gdk-pixbuf gobject-introspection libnotify gtk3 ];
-  pythonPath = with pythonPackages; [
+  pythonPath = with python3Packages; [
```

Comment:
We set `pythonPath` here, not `propagatedBuildInputs`, because this will never be imported anyways?

Reply:
Sorry, I don't quite understand your question. caffeine-ng imports those packages somewhere (because it fails without them) but importing caffeine-ng seems strange for me, since it is an application intended for end-users and there AFAIK there is no documentation for doing so.

Reply:
I don't understand why `pythonPath` is set here - I'd put most of these dependencies inside `propagatedBuildInputs` as described in https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies, except `setuptools` `setuptools_scm` `wheel`, I'm not sure if those are needed at all.

Reply:
It seems `setuptools_scm` is needed since without it, the build fails.

Reply:
As far as I understand from https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters the difference between `pythonPath` and `propagatedBuildInputs` is that `pythonPath` does not propagate dependencies. IMHO this is preferable here, since a dependency of a tray application does (read as: should) not care about accessing the dependencies of it directly.

Reply:
There seems to have been discussion around the preferable behavior here, see e.g. https://gist.github.com/FRidh/c3e974e34eac738405af699786b462d5, #24128, #1819, #11423. This seems mostly non-active (i.e. pre-2017) but unresolved, that is, I could not find any results from this discussion.
A quick `grep`through nixpkgs seems to have an about 50-50-split between using `pythonPath` and `propagatedBuildInputs`.
I'd be happy to change `pythonPath` to `propagatedBuildInputs` if this is the preferred option, however, since I did not find a clear consensus, I would lean to leaving it as it is for now.

Reply:
I'm fine with both. `setuptools_scm` can't be moved to `buildInputs`? This shouldn't be used after building…

Reply:
You are right, this should be possible. Will change that. Thanks.